### PR TITLE
yoshih/submodule sst-jucegui

### PR DESCRIPTION
**Pull request by shih1**
Submodule https://github.com/surge-synthesizer/sst-jucegui into `surge`. 

Testing
- clean build and ran synth + fx, twiddled a few knobs, switched menus

### CODE REVIEW

File: src/surge-fx/CMakeLists.txt [hunk 1]
```diff
@@ -67,7 +67,7 @@ target_sources(${PROJECT_NAME} PRIVATE
 target_link_libraries(${PROJECT_NAME} PRIVATE
   surge::surge-common
   surge-fx-binary
-  surge-juce
```
Comment (shih1) [inline src/surge-fx/CMakeLists.txt:70]: i'm not entirely sure this is ok, it works but I don't fully understand the implications of this change. 

CMake is not my strongest language.

**Comment by baconpaul** [comment]
My guess is the failure is unrelated to this and is related to Ubuntu latest moving to 24 nd me needing to update the prepare for juce action 

but the sst jucegui should have an include on use juce I bet.

**Comment by baconpaul** [comment]
yeah definitely the case that the failure isn't yours

but the inclusion in lib.cmake should be wrapped in a  `if (NOT SURGE_SKIP_JUCE_FOR_RACK)` block

i'll try and get the pipelines fixed then will re-run this if i do over in #7925

File: src/common/CMakeLists.txt [hunk 2]
```diff
@@ -33,6 +33,7 @@ surge_add_lib_subdirectory(sst/sst-cpputils)
 surge_add_lib_subdirectory(sst/sst-plugininfra)
 surge_add_lib_subdirectory(sst/sst-filters)
 surge_add_lib_subdirectory(sst/sst-waveshapers)
+surge_add_lib_subdirectory(sst/sst-jucegui)
```
Comment (baconpaul) [inline src/common/CMakeLists.txt:36]: This inclusion requires a `if (NOT SURGE_SKIP_JUCE_FOR_RACK)` / `endif()` block surrounding it. Thanks!

**Review by baconpaul — CHANGES_REQUESTED** [review]

File: src/common/CMakeLists.txt (hunk 2 above)
Comment (shih1) [reply src/common/CMakeLists.txt:36]: i moved it down into the existing block

**Pull request merged**